pkg/cmd/infra/router/template.go

```diff
@@ -193,11 +205,75 @@ func (o *TemplateRouterOptions) Validate() error {
 
 // Run launches a template router using the provided options. It never exits.
 func (o *TemplateRouterOptions) Run() error {
+	var reloadCallbacks []func()
+	switch {
+	case o.MetricsType == "haproxy" && len(o.ListenAddr) > 0:
+		if len(o.StatsUsername) == 0 || len(o.StatsPassword) == 0 {
+			glog.Warningf("Metrics were requested but no username or password has been provided - the metrics endpoint will not be accessible to prevent accidental security breaches")
+		}
+		// Exposed to allow tuning in production if this becomes an issue
+		var timeout time.Duration
+		if t := util.Env("ROUTER_METRICS_HAPROXY_TIMEOUT", ""); len(t) > 0 {
+			d, err := time.ParseDuration(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_TIMEOUT is not a valid duration: %v", err)
+			}
+			timeout = d
+		}
+		// Exposed to allow tuning in production if this becomes an issue
+		var baseScrapeInterval time.Duration
+		if t := util.Env("ROUTER_METRICS_HAPROXY_BASE_SCRAPE_INTERVAL", ""); len(t) > 0 {
+			d, err := time.ParseDuration(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_BASE_SCRAPE_INTERVAL is not a valid duration: %v", err)
+			}
+			baseScrapeInterval = d
+		}
+		// Exposed to allow tuning in production if this becomes an issue
+		var serverThreshold int
+		if t := util.Env("ROUTER_METRICS_HAPROXY_SERVER_THRESHOLD", ""); len(t) > 0 {
+			i, err := strconv.Atoi(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_SERVER_THRESHOLD is not a valid integer: %v", err)
+			}
+			serverThreshold = i
+		}
+		// Exposed to allow tuning in production if this becomes an issue
+		var exported []int
+		if t := util.Env("ROUTER_METRICS_HAPROXY_EXPORTED", ""); len(t) > 0 {
+			for _, s := range strings.Split(t, ",") {
+				i, err := strconv.Atoi(s)
+				if err != nil {
+					return errors.New("ROUTER_METRICS_HAPROXY_EXPORTED must be a comma delimited list of column numbers to extract from the HAProxy configuration")
+				}
+				exported = append(exported, i)
+			}
+		}
+		_, err := haproxy.NewPrometheusCollector(haproxy.PrometheusOptions{
+			// Only template router customizers who alter the image should need this
+			ScrapeURI: util.Env("ROUTER_METRICS_HAPROXY_SCRAPE_URI", ""),
+			// Only template router customizers who alter the image should need this
+			PidFile:            util.Env("ROUTER_METRICS_HAPROXY_PID_FILE", ""),
+			Timeout:            timeout,
+			ServerThreshold:    serverThreshold,
+			BaseScrapeInterval: baseScrapeInterval,
+			ExportedMetrics:    exported,
+		})
+		if err != nil {
+			return err
+		}
+		//reloadCallbacks = append(reloadCallbacks, e.CollectNow)
```

Comment:
Originally I thought we wanted this, but it didn't make the experience / logging better.  So I disabled it - I can add a comment that this is disabled pending a resolution of maybe moving to a newer haproxy version.